app/src/main/java/com/example/instagramclone/fragments/message_screen/compose/MessagesComposable.kt

```diff
@@ -0,0 +1,126 @@
+import androidx.navigation.NavHostController
+import coil.compose.AsyncImage
+import com.example.domain.entities.Message
+import com.example.instagramclone.R.drawable
+import com.example.instagramclone.fragments.compose.components.GreyTextField
+import com.example.instagramclone.fragments.message_screen.MessageScreenState.Success
+import com.example.instagramclone.fragments.message_screen.MessageViewModel
+import com.example.instagramclone.fragments.ui.theme.EtGrey2
+import com.example.instagramclone.fragments.ui.theme.FBlue
+
+@OptIn(ExperimentalLifecycleComposeApi::class, ExperimentalAnimationApi::class)
+@Composable
+fun MessagesComposable(
+    messageViewModel: MessageViewModel = hiltViewModel(),
+    navController: NavHostController
+) {
+    Scaffold(topBar = {
+        MessagesTopBar(navController = navController)
+    }) {
+        val uiState by messageViewModel.uiState.collectAsStateWithLifecycle()
+
+        AnimatedContent(targetState = uiState) { dataState ->
+            when (dataState) {
+                is Success -> {
+                    MessageScreen(
+                        modifier = Modifier.padding(it),
+                        messageList = dataState.messageList
+
+                    )
+                }
+                else -> Unit
+            }
+        }
+    }
+}
+
+@Composable
+fun MessageScreen(modifier: Modifier = Modifier, messageList: List<Message>) {
+    Column(modifier = modifier) {
+        LazyColumn() {
+            item {
+                MesssageUtilSection()
+            }
+            items(
+                messageList
+            ) { messageItem ->
+                MessageItem(messageItem)
+            }
+        }
+    }
+}
+
+@Composable
+fun MesssageUtilSection() {
+    GreyTextField(
+        modifier = Modifier
+            .padding(12.dp)
+            .fillMaxWidth(),
+        text = "",
+        hintText = "Search",
+        backgroundColor = MaterialTheme.colors.EtGrey2,
+        onValueChange = {
+        }
+    )
+    AsyncImage(
+        modifier = Modifier
+            .padding(horizontal = 22.dp, vertical = 8.dp)
+            .size(84.dp)
+            .clip(CircleShape),
+        contentScale = ContentScale.Crop,
+        placeholder = painterResource(id = drawable.bg_placeholder),
+        model = "https://picsum.photos/id/${19}/500/500",
+        contentDescription = "profile image"
+    )
+    Text(
+        modifier = Modifier
+            .padding(horizontal = 32.dp),
+        text = "Your Note"
+    )
+    Row(
+        modifier = Modifier
+            .fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        Text(
+            modifier = Modifier.padding(all = 12.dp),
+            text = "Messages",
+            fontWeight = FontWeight.Bold,
+            fontSize = 16.sp
+        )
+        Text(
+            modifier = Modifier.padding(all = 12.dp),
+            text = "Requests",
+            fontSize = 16.sp,
```

Comment:
It's best to use the [4-point grid system](https://medium.com/@aratidube12lns/4-point-grid-system-for-more-consistent-interface-design-efea81dea3f3#:~:text=What%20is%20the%204%2Dpoint,padding%2C%20margins%20and%20line%20heights.) wherever possible.

And let's extract all string resources in this file